.github/workflows/ci.yaml

```diff
@@ -19,6 +19,9 @@ jobs:
           - 3.8
           - 3.9
     runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash -l {0}
```

Comment:
[The default](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell) is

```
bash --noprofile --norc -eo pipefail {0}
```

so this change effectively removes the `-eo pipefail` options. We want to keep them.

Hopefully by switching to a login shell (`-l`) and thus sourcing the profile and rc files we don't bring in other unexpected changes besides whatever Conda added. This is why I just explicitly set up `PATH` instead.

Reply:
The problem with `PATH` is it calls for a separate script for Windows if we want to support that. But we might have to do that regardless, to some extent.

I'll:

1. Add `-eo pipefail` to this default shell
2. Add commands to write the contents of `~/.profile` and `~/.bashrc` so we can make sure nothing unwanted is in there.

Reply:
Sounds good!

Reply:
The [new CI run](https://github.com/nextstrain/cli/actions/runs/1960723783) shows that there is no `~/.bashrc` on these runners. Here are the contents of `~/.profile` on each os:

ubuntu-latest:

```
# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$('/usr/share/miniconda/bin/conda' 'shell.bash' 'hook' 2> /dev/null)"
if [ $? -eq 0 ]; then
    eval "$__conda_setup"
else
    if [ -f "/usr/share/miniconda/etc/profile.d/conda.sh" ]; then
        . "/usr/share/miniconda/etc/profile.d/conda.sh"
    else
        export PATH="/usr/share/miniconda/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda initialize <<<


  # ----------------------------------------------------------------------------
  # Conda Setup Action: Basic configuration
  set -eo pipefail
  # Conda Setup Action: Custom activation
  conda activate "test"
  # ----------------------------------------------------------------------------
```

macos-latest:

```
# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$('/usr/local/miniconda/bin/conda' 'shell.bash' 'hook' 2> /dev/null)"
if [ $? -eq 0 ]; then
    eval "$__conda_setup"
else
    if [ -f "/usr/local/miniconda/etc/profile.d/conda.sh" ]; then
        . "/usr/local/miniconda/etc/profile.d/conda.sh"
    else
        export PATH="/usr/local/miniconda/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda initialize <<<


  # ----------------------------------------------------------------------------
  # Conda Setup Action: Basic configuration
  set -eo pipefail
  # Conda Setup Action: Custom activation
  conda activate "test"
  # ----------------------------------------------------------------------------
```

Mostly all conda stuff which is wanted. The conda action also adds `set -eo pipefail`, but good to have it defined in our default shell in case they remove it here.

Reply:
Ah, nice to see them restore `-eo pipefail` since they document switching the shell to just `bash -l`.